AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -195,6 +193,36 @@ actor EndpointClient: EndpointClientBehaviour {
                                    endpointId: endpointProfile.endpointId,
                                    endpointRequest: endpointRequest)
     }
+    
+    func getPublicEndpoint() -> PinpointClientTypes.PublicEndpoint {
+        let endpointProfile = currentEndpointProfile()
+        let channelType = getChannelType(from: endpointProfile)
+        let effectiveDate = getEffectiveDateIso8601FractionalSeconds(from: endpointProfile)
+        let optOut = getOptOut(from: endpointProfile)
+        let publicEndpoint = PinpointClientTypes.PublicEndpoint(
+            address: endpointProfile.endpointId,
```

Comment:
Are you sure you need to map `endpointProfile.endpointId` to `publicEndpoint.address` for this operation?

I'm asking because on the update input, we map the `address`  field to the `deviceToken` instead, as the [legacy Pinpoint SDK does it](https://github.com/aws-amplify/aws-sdk-ios/blob/main/AWSPinpoint/AWSPinpointEndpointProfile.m#L82).

Reply:
good catch, updated to `endpointProfile.deviceToken`.  In legacy SDK, [it appears to be the device token](https://github.com/aws-amplify/aws-sdk-ios/blob/main/AWSPinpoint/AWSPinpointEventRecorder.m#L1172)